sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GlobalAddressResolver.java

```diff
@@ -94,69 +97,142 @@ public GlobalAddressResolver(
+                                                        .collect(Collectors.toList());
+
+                                                return Mono.just(new ImmutablePair<String, List<PartitionKeyRangeIdentity>>(containerLink, pkrs));
+                                            })
+                                            .flatMapMany(containerLinkToPkrs -> this.resolveAddressesPerCollection(
+                                                            containerLinkToPkrs.left,
+                                                            collection,
+                                                            containerLinkToPkrs.right,
+                                                            proactiveContainerInitConfig,
+                                                            hint
+                                                    )
+                                            ).flatMap(collectionToAddresses -> {
+                                                ImmutablePair<String, DocumentCollection> containerLinkToCollection
+                                                        = collectionToAddresses.left;
+                                                AddressInformation addressInformation =
+                                                        collectionToAddresses.right;
+
+                                                Map<String, Integer> containerLinkToMinConnectionsMap = ImplementationBridgeHelpers
+                                                        .CosmosContainerProactiveInitConfigHelper
+                                                        .getCosmosContainerIdentityAccessor()
+                                                        .getContainerLinkToMinConnectionsMap(proactiveContainerInitConfig);
+
+                                                int connectionsPerReplicaCountForContainer = containerLinkToMinConnectionsMap
+                                                        .getOrDefault(
+                                                                containerLinkToCollection.left,
+                                                                Configs.getMinConnectionPoolSizePerEndpoint()
+                                                        );
+
+                                                return this.openConnectionInternal(
+                                                        addressInformation,
+                                                        containerLinkToCollection.getRight(),
+                                                        proactiveContainerInitConfig,
+                                                        connectionsPerReplicaCountForContainer,
+                                                        hint
+                                                );
+                                            });
+                                }), Configs.getCPUCnt(), Configs.getCPUCnt());
```

Comment:
This could help limit the no. of containers on which concurrent get address requests are performed.